### ISSUE TITLE
Incorrect use of action to set custom variable causes companion to lock-up

### ISSUE THREAD

**Issue by champagj** [bitfocus/companion#1848]
**Describe the bug**
When using _internal: Set custom variable value_ instead of  _internal: Set custom variable expression_ action to set a value with an expression, companion locks up.

I incorrectly chose set value to update a custom variable with an expression 

**To Reproduce**
Create a custom variable:  jf_segment
Create a button to increment the value of jf_segment by 1
Select Action _internal: Set custom variable value_ with Value _$(internal:custom_jf_segment) + 1_
Press button. 
Companion is frozen

**Expected behavior**
While this is caused by user error, it is fairly likely to happen as the actions are similar and relate to similar functions
The section on _getting-started#3-custom-variables_ mentions that invalid expressions would be ignored. 
Similar test/check should be done on _value set_ to ensure they are valid as well

**Screenshots**
![image](https://user-images.githubusercontent.com/47355785/148630648-5f034983-d5e6-426a-b7a1-e5922756c999.png)

**Environment**
Companion Pi 2.2.0 running built 3694

**Comment by krocheck** [comment]
@champagj are you able to get at the console to get a readout on the error?  I tried to reproduce on Win and it simply set the value of the variable as expected to the plain text.  There's a try-catch on action execution intentionally as a catch all for potential edge cases like this, so its unclear at face value what's going on here (at least to me).

![image](https://user-images.githubusercontent.com/749812/148643976-e6156205-c642-4585-a5d2-ed981f6f6007.png)
![image](https://user-images.githubusercontent.com/749812/148643986-fa03452c-39ea-4076-b4a7-d4cbe35a805c.png)

**Comment by champagj** [comment]
@krocheck when you say console, do you mean the log with debug selected?

I ran the test again and captured the log. Offending button is 1.8 and did not even register on the log before locking up.

**Log (debug mode)**
![image](https://user-images.githubusercontent.com/47355785/148649596-3996de23-c264-430c-9f55-3c57ae9c8426.png)

I also open an ssh session with the Raspberry Pi but there is no verbose there.. 
I am not sure where else to go to get a console output, if someone can help me out here, I'll do the test again.

Thanks,

**Comment by krocheck** [comment]
@champagj Understood.  Are you running using the `yarn headless` command?  Running `yarn dev-headless` will give you a verbose output.

**Comment by champagj** [comment]
Got it. Thanks. 
Left for the weekend. I'll follow-up early next week

**Comment by champagj** [comment]
> @champagj Understood. Via SSH are you running using the `yarn headless` command? Running `yarn dev-headless` will give you a verbose output.

not using _yarn headless_ command to launch companion. 
Used the _manual install_ method in the wiki for install. Companion running as a service at boot from the service file I guess that the launch command is:
_node /home/pi/companion/headless_ip.js 0.0.0.0_

I can stop the service/companion via SSH and then I get companion to start verbose with _yarn dev-headless_ but does not bind to an interface and does not load config. I tried a few different commands / options to get the debug output but no success.
I need a bit of help to manually launch companion via SSH in dev mode..

**Comment by Julusian** [comment]
@krocheck Ive just given this a try and all the output I get is:
```
  lib/action Running action {
  id: 'wIeknG-tZ',
  label: 'bitfocus-companion:custom_variable_set_value',
  instance: 'bitfocus-companion',
  action: 'custom_variable_set_value',
  options: { name: 'aaaa', value: '$(internal:custom_aaaa) +1' },
  delay: 0
} +0ms
  lib/variable Set value "aaaa":$(internal:custom_aaaa) +1 +5s
  lib/variable Variable $(internal:custom_aaaa) is "$(internal:custom_aaaa) +1" +0ms
  lib/bank variable changed in bank 1.27 +2ms
  lib/server_tcp bank_bg send to all open sockets {"type":"bank_bg_change","page":1,"bank":27,"red":0,"green":0,"blue":0} +2ms
```

Yeah, its getting lost in an infinite inside of 'variable_parse' replacing the variable with itself..

**Comment by krocheck** [comment]
Yeah, I see it.  I'm thinking we have an array to store the keys in as they're parsed and check each key to see if it comes through a 2nd time.

**Comment by Julusian** [comment]
Hopefully this will be resolved without side effects in #1867
